test/auto_shorter_deprecated.cc

```diff
@@ -0,0 +1,67 @@
+#include "test.h"
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "baldr/graphreader.h"
+#include "baldr/rapidjson_utils.h"
+#include "loki/worker.h"
+#include "odin/worker.h"
+#include "thor/worker.h"
+#include "tyr/serializers.h"
+#include "valhalla/worker.h"
+#include <boost/property_tree/ptree.hpp>
+
+using namespace valhalla;
+using namespace valhalla::baldr;
+using namespace valhalla::thor;
+using namespace valhalla::odin;
+using namespace valhalla::loki;
+using namespace valhalla::tyr;
+
+namespace {
+
+TEST(AutoShorterDeprecated, test_auto_shorter_replace) {
+  // Test if auto_shorter will be converted to auto with shortest: true
+  Api request;
+  std::string request_str =
+      R"({"locations":[{"lat":52.114622,"lon":5.131816},{"lat":52.048267,"lon":5.074825}],"costing":"auto_shorter"})";
+  ParseApi(request_str, Options::route, request);
+
+  EXPECT_EQ(Costing_Enum_Name(request.options().costing()), "auto");
+  EXPECT_EQ(request.options().costing_options(static_cast<int>(auto_)).shortest(), true);
+}
+
+TEST(AutoShorterDeprecated, test_auto_shorter_shortest_false) {
```

Comment:
test name is stupid. same below. will change before it'll be merged

Reply:
i suggest we remove this whole test suite and instead just add the `ParseApi` part as individual test cases in your other new test file that is specific for testing shortest. that way we can test that the parsing override works and gives us an `auto` cost with `shortest` set to true. let me see if i can do that quickly